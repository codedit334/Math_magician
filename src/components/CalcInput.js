import React from 'react';
import '../styles/calcInput.css';

export default function CalcInput(props) {
  const res = { props };
  return (
    <div className="calcinput">
      <input
        data-testid="CalcInput"
        className="calcinput-display"
        disabled
        value={`${res.props.resObj.total || ''}${
          res.props.resObj.operation || ''
        }${res.props.resObj.next || ''}`}
        placeholder={res.props.resObj.next || res.props.resObj.total}
      />
    </div>
  );
}
