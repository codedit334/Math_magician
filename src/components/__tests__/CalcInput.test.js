import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import CalcInput from '../CalcInput';

test('Sould render CalcInput Component', () => {
  const testObj = {
    total: 9,
    next: null,
    operation: null,
  };
  render(<CalcInput resObj={testObj} />);
  const calcInputElement = screen.getByTestId('CalcInput');
  expect(calcInputElement).toBeInTheDocument();
  expect(calcInputElement.value).toBe('9');

  const tree = renderer.create(<CalcInput resObj={testObj} />).toJSON();
  expect(tree).toMatchSnapshot();
});
