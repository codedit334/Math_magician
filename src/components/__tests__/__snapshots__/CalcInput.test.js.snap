// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sould render CalcInput Component 1`] = `
<div
  className="calcinput"
>
  <input
    className="calcinput-display"
    data-testid="CalcInput"
    disabled={true}
    placeholder={9}
    value="9"
  />
</div>
`;
